src/fs.ts

```diff
@@ -99,7 +99,7 @@ export class FileSystemUpdater {
 	 */
 	private concurrencyLimit = new Semaphore(100);
 
-	constructor(private remoteFs: FileSystem, private inMemoryFs: InMemoryFileSystem) {}
+	constructor(public remoteFs: FileSystem, public inMemoryFs: InMemoryFileSystem) {}
```

Comment:
I don't like this change. FileSystemUpdater has a dependency on `InMemoryFileSystem` and `ProjectManager` has a dependency on `InMemoryFileSystem`. By making them public and not explicitly passing them to `ProjectManager` you make the dependencies unclear. For example, ProjectManager now has access to the remote file system, but should only interact with the Updater.